mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param await_registration_for: Number of seconds to wait for the model version to finish
+                            being created and is in ``READY`` status. By default, the function
+                            waits for five minutes. Specify 0 or None to skip waiting.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :return: A :py:class:`ModelInfo <mlflow.models.model.ModelInfo>` instance that contains the
+             metadata of the logged model.
+
+    .. code-block:: python
+        :caption: Example
+
+        from pyspark.ml import Pipeline
+        from pyspark.ml.classification import LogisticRegression
+        from pyspark.ml.feature import HashingTF, Tokenizer
+
+        training = spark.createDataFrame(
+            [
+                (0, "a b c d e spark", 1.0),
+                (1, "b d", 0.0),
+                (2, "spark f g h", 1.0),
+                (3, "hadoop mapreduce", 0.0),
+            ],
+            ["id", "text", "label"],
+        )
+        tokenizer = Tokenizer(inputCol="text", outputCol="words")
+        hashingTF = HashingTF(inputCol=tokenizer.getOutputCol(), outputCol="features")
+        lr = LogisticRegression(maxIter=10, regParam=0.001)
+        pipeline = Pipeline(stages=[tokenizer, hashingTF, lr])
+        model = pipeline.fit(training)
+        mlflow.spark.log_model(model, "spark-model")
+    """
+    validate_env_vars()
+    run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+    run_root_artifact_uri = mlflow.get_artifact_uri()
+    remote_model_path = None
+    if _should_use_mlflowdbfs(run_root_artifact_uri):
+        remote_model_path = append_to_uri_path(
+            run_root_artifact_uri, artifact_path, _JOHNSNOWLABS_MODEL_PATH_SUB
+        )
+        mlflowdbfs_path = _mlflowdbfs_path(run_id, artifact_path)
+        with databricks_utils.MlflowCredentialContext(
+                get_databricks_profile_uri_from_artifact_uri(run_root_artifact_uri)
+        ):
+            try:
+                _unpack_and_save_model(spark_model, mlflowdbfs_path)
+
+            except Exception as e:
+                raise MlflowException("failed to save spark model via mlflowdbfs") from e
```

Comment:
Please update the Exception message and specify an ErrorCode for the Exception

Reply:
It's based on [this line in mlflow.spark.py](https://github.com/mlflow/mlflow/blob/616858977d3964933839141967e13124491acd0d/mlflow/spark.py#L244)

Updated the message, but not sure which ErrorCode to use here, since I try to stay in line with the `spark` flavor which seems to use the default message as well